packages/p2p-v2/src/hooks/__tests__/useQueryString.spec.tsx

```diff
@@ -0,0 +1,226 @@
+                    })
+                }
+            >
+                Switch counterparties
+            </button>
+            <button data-testid='go_back_btn' onClick={() => deleteQueryString('tab')}>
+                Go back
+            </button>
+            {!queryString.get('tab') && <h1>My profile</h1>}
+            {queryString.get('tab') === 'Stats' && <h1>Stats tab</h1>}
+            {queryString.get('tab') === 'Counterparties' && <h1>Counterparties tab</h1>}
+            {queryString.get('tab') === 'Payment methods' && <h1>Payment methods tab</h1>}
+            {queryString.get('form') === 'Payment method form' && <h1>Payment method form</h1>}
+        </div>
+    );
+};
+
+const MockHome = () => <h1>Home</h1>;
+
+const MockAppContent = () => {
+    return (
+        <Switch>
+            <Route component={MockHome} exact path='/' />
+            <Route component={MockBuySell} path='/mock-buy-sell' />
+            <Route component={MockMyProfile} path='/mock-my-profile' />
+        </Switch>
+    );
+};
+
+const mockReplace = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useHistory: () => ({
+        push: jest.fn(),
+        replace: mockReplace,
+    }),
+}));
+
+let windowLocationSpy: jest.SpyInstance<Location, []>;
+
+describe('useQueryString', () => {
+    beforeEach(() => {
+        windowLocationSpy = jest.spyOn(window, 'location', 'get');
+    });
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+    test('setting new query strings', () => {
+        const history = createMemoryHistory();
+        history.push('/mock-my-profile');
+        render(
+            <Router history={history}>
+                <MockAppContent />
+            </Router>
+        );
+        expect(screen.getByText('My profile')).toBeInTheDocument();
+        expect(screen.queryByText('Payment methods tab')).not.toBeInTheDocument();
+        expect(screen.queryByText('Stats tab')).not.toBeInTheDocument();
+
+        const btn = screen.getByTestId('switch_payment_methods_btn');
+        const originalLocation = window.location;
+        const queryChangeEventSpy = jest.spyOn(window, 'dispatchEvent');
+        windowLocationSpy.mockImplementation(() => ({
+            ...originalLocation,
+            href: 'http://localhost/my-profile?tab=Payment+methods',
+            pathname: 'my-profile',
+            search: '?tab=Payment+methods',
+        }));
+        btn.click();
+        expect(mockReplace).toHaveBeenLastCalledWith({
+            pathname: 'my-profile',
+            search: 'tab=Payment+methods',
+        });
+        expect(mockReplace).toHaveBeenCalled();
+        expect(queryChangeEventSpy).toHaveBeenCalledWith(expect.any(Event));
+        expect(queryChangeEventSpy.mock.calls[0][0].type).toBe('queryChange');
+        expect(screen.getByText('Payment methods tab')).toBeInTheDocument();
+
+        const statsBtn = screen.getByTestId('switch_stats_btn');
```

Comment:
lets use getByRole instead of getByTestId 🙏 

```suggestion
        const statsBtn = screen.getByRole('button', { name: 'Switch stats' });
```